ambry-commons/src/test/java/com.github.ambry.commons/ResponseHandlerTest.java

```diff
@@ -106,29 +108,67 @@ public ReplicaId getLastReplicaID() {
   public void basicTest() {
     DummyMap map = new DummyMap();
     ResponseHandler handler = new ResponseHandler(map);
-    handler.onRequestResponseException(new MockReplicaId(), new SocketException());
+
```

Comment:
minor: It would probably be shorter (and easier to understand) if you had a `Map<Object, List<ReplicaEventType>> with all the objects that are supposed to trigger events with the value being the `ReplicaEventType`s that should be fired. This should work for the ones that should be "unrecognized" too.